ocs_ci/utility/kms.py

```diff
@@ -1655,24 +1656,30 @@ def get_key_info_ciphertrust(self, key_id):
         else:
             return json.loads(out)
 
-    def get_key_list_ciphertrust(self):
+    def get_key_list_ciphertrust(self, limit=100):
         """
         Lists all keys in CipherTrust Manager
 
+        Args:
+            limit (int): number of entries to limit the results
+
         Returns:
             (list): list containing the IDs of the keys
 
         """
         key_id_list = []
-        cmd = "ksctl keys list"
-        out = subprocess.check_output(shlex.split(cmd))
-        json_out = json.loads(out)
-        if json_out["total"] == 0:
-            raise NotFoundError("No keys found")
-        else:
-            for key in json_out["resources"]:
-                key_id_list.append(key["id"])
-            return key_id_list
+        total = None
+        while len(key_id_list) != total:
```

Comment:
can we remove this while loop, didn't see use case ( 
 raising NotFoundError if total = 0 otherwise returning key_id_list) where its iterating over loop.

Reply:
ah, return key_id_list supposed to be indented out of the loop.

So this condition should iterate over the command to get e.g. all 381 keys in kind of paging mechanism to get every query max 100 keys.

Thanks for noticing this.